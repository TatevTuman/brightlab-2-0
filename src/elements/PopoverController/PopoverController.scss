@import "static/assets/styles/mixins";

.controller {
  position: relative;
  cursor: pointer;

  &-button {
    position: absolute;
    left: toRem(100);
    display: flex;
    justify-content: center;
    align-items: center;
    width: toRem(34);
    height: toRem(34);
    background: #F6F7FF;
    border: 0.5px solid #D7D7E8;
    box-sizing: border-box;
    border-radius: toRem(10);
    font-size: toRem(14);
    color: var(--purple);

    & > div {
      display: none;
      position: absolute;
      bottom: toRem(50);
      background: rgba(62, 66, 223, 0.09);

      &:after {
        display: block;
      }
    }

    &:hover {
      background: rgba(62, 66, 223, 0.09);

      div {
        display: block;
      }
    }
    &[data-bottom='true'] {

      & > div {
        bottom: toRem(-160);

        &:after {
          bottom: toRem(140);
        }
      }
    }
  }
}