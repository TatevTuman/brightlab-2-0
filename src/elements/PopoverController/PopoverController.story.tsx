import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import PopoverController, { PopoverControllerProps } from './PopoverController'

export default {
  title: 'Elements/Popover–°ontroller',
  component: PopoverController,
  argTypes: {}
} as Meta

const Template: Story<PopoverControllerProps> = args => <PopoverController {...args} />

export const ExamplePopoverController = Template.bind({})
