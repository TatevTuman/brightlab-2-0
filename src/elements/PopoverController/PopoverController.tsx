import React, { memo, useState } from 'react'
import { Popover } from '@elements'
import './PopoverController.scss'

export interface PopoverControllerProps {
  bottom?: boolean
  onClick?(e: MouseEvent): void
}

const PopoverController: React.FC<PopoverControllerProps> = props => {
  const { bottom } = props
  const [open, setOpen] = useState(false)
  return (
    <div className={'controller'}>
      <div className={'controller-button'} onClick={() => setOpen(!open)} data-bottom={bottom}>
        <Popover open={open} />
      </div>
    </div>
  )
}
PopoverController.defaultProps = {
  bottom: false
}

export default memo(PopoverController)
