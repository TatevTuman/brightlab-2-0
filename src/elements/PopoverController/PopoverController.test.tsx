import React from 'react'
import { cleanup, render, waitFor } from '@testing-library/react'
import PopoverController from './PopoverController'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('Popover', () => {
  it('renders correctly', async () => {
    const { container } = render(<PopoverController />)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
  })
})
