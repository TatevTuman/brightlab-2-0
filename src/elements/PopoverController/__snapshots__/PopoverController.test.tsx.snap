// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover renders correctly 1`] = `
<div>
  <div
    class="controller"
  >
    <div
      class="controller-button"
      data-bottom="false"
    >
      <div
        class="popover"
        data-open="false"
      >
        <div
          class="popover-list"
        >
          <div
            class="popover-list-option"
          >
            <div>
              Michael Stevenson
            </div>
            <div>
              Figma Corporate. Presentation and Sales...
            </div>
            <div>
              <div>
                <i
                  class="icon"
                >
                  <test-file-stub
                    name="clock"
                  />
                </i>
              </div>
              <div>
                8:00 - 9:30
              </div>
            </div>
          </div>
          <div
            class="popover-list-option"
          >
            <div>
              Michael Stevenson
            </div>
            <div>
              Figma Corporate. Presentation and Sales...
            </div>
            <div>
              <div>
                <i
                  class="icon"
                >
                  <test-file-stub
                    name="clock"
                  />
                </i>
              </div>
              <div>
                8:00 - 9:30
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
